spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/KafkaMessageChannelBinder.java

```diff
@@ -194,49 +142,30 @@ public KafkaProducerProperties getExtendedProducerProperties(String channelName)
 	}
 
 	@Override
-	protected MessageHandler createProducerMessageHandler(final String destination,
+	protected MessageHandler createProducerMessageHandler(final ProducerDestination destination,
 															ExtendedProducerProperties<KafkaProducerProperties> producerProperties) throws Exception {
-
-		KafkaTopicUtils.validateTopicName(destination);
-		createTopicsIfAutoCreateEnabledAndAdminUtilsPresent(destination, producerProperties.getPartitionCount());
-		Collection<PartitionInfo> partitions = getPartitionsForTopic(destination, producerProperties.getPartitionCount());
-
+		final DefaultKafkaProducerFactory<byte[], byte[]> producerFB = getProducerFactory(producerProperties);
+		Collection<PartitionInfo> partitions = ((KafkaTopicProvisioner)provisioningProvider)
```

Comment:
That's what I mean in the https://github.com/spring-cloud/spring-cloud-stream/pull/804/files#r101065893.

As I said: just my fear.
If you are fine with this, we can go ahead.